### PR TITLE
[mac] tracking supported channel mask

### PR DESCRIPTION
**Pull request by abtink**
This commit contains the following changes:
- Adds new logic to `Mac` to track supported channel mask along
  get/set methods.
- Adds public OT APIs `otLinkGetSupportedChannelMask()` and
 `otLinkSetSupportedChannelMask()` to get/set the channel mask.
- Ensures supported channel mask from `Mac` is included in Dataset
  TLVs when a local Dataset is generated.
- Ensures the channel mask at `Mac` gets updated when a new Dataset
  configuration is applied.
- Adds `OT_CHANGED` notification for signaling channel mask changes.
- Updates `NcpBase` and `PHY_CHAN_SUPPORTED` spinel property get/set
  handlers to adopt the new public APIs.
- Updates `NcpBase` to emit async `PHY_CHAN_SUPPORTED` updated on
  OT_CHANGED channel mask notification.

### CODE REVIEW

File: include/openthread/link.h [hunk 1]
```diff
@@ -189,14 +189,38 @@ OTAPI uint8_t OTCALL otLinkGetChannel(otInstance *aInstance);
  * @param[in]  aChannel    The IEEE 802.15.4 channel.
  *
  * @retval  OT_ERROR_NONE           Successfully set the channel.
- * @retval  OT_ERROR_INVALID_ARGS   If @p aChnanel is not in the range [11, 26].
+ * @retval  OT_ERROR_INVALID_ARGS   If @p aChnanel is not in the range [11, 26] or is not in the supported channel mask.
```
Comment (jwhui) [inline include/openthread/link.h]: "aChnanel" -> "aChannel"

File: src/core/mac/mac.hpp [hunk 2]
```diff
@@ -643,6 +646,22 @@ class Mac : public InstanceLocator
      */
     otError ReleaseRadioChannel(void);
 
+    /**
+     * This method return the supported channel mask.
```
Comment (jwhui) [inline src/core/mac/mac.hpp]: "return" -> "returns"

**Comment by Ahmedkalnaggar** [comment]
@abtink Could you review the replay from Maria here :  [TSTBED-106](https://threadgroup.atlassian.net/browse/TSTBED-106) ?

**Comment by jwhui** [comment]
@Ahmedkalnaggar, this PR is closed, please keep the discussion in #2985.

**Pull request merged**